pkg/api/validation/validation.go

```diff
@@ -2444,6 +2421,8 @@ func validatePodAffinityTerm(podAffinityTerm api.PodAffinityTerm, allowEmptyTopo
 	}
 	if len(podAffinityTerm.TopologyKey) != 0 {
 		allErrs = append(allErrs, unversionedvalidation.ValidateLabelName(podAffinityTerm.TopologyKey, fldPath.Child("topologyKey"))...)
+	} else {
```

Comment:
It does not make sense to have both 
```        
if !allowEmptyTopologyKey && len(podAffinityTerm.TopologyKey) == 0 {
                allErrs = append(allErrs, field.Required(fldPath.Child("topologyKey"), "can only be empty for PreferredDuringScheduling pod anti affinity"))
        }
```
where it says that `can only be empty for PreferredDuringScheduling pod anti affinity`.

Also 
```
len(podAffinityTerm.TopologyKey) != 0 
```
for all cases what you did.